src/python/pants/python/pex_build_util.py

```diff
@@ -427,14 +427,23 @@ def _prepare_inits(self) -> Set[str]:
     def set_emit_warnings(self, emit_warnings):
         self._builder.info.emit_warnings = emit_warnings
 
+    def _set_major_minor_interpreter_constraint_for_ipex(
+        self, info: PexInfo, identity: PythonIdentity,
+    ) -> PexInfo:
```

Comment:
Thanks for type hints :)

Reply:
This was some really funky code that serialized a `PexInfo` to json and back :D type hints make it easier to think!